rindti/models/dti/classification.py

```diff
@@ -12,11 +13,14 @@ class ClassificationModel(BaseModel):
     """Model for DTI prediction as a class problem."""
 
     def __init__(self, **kwargs):
-        kwargs = super().__init__(**kwargs)
-        self._determine_feat_method(kwargs["feat_method"], kwargs["prot"]["hidden_dim"], kwargs["drug"]["hidden_dim"])
-        self.prot_encoder = Encoder(**kwargs["prot"])
-        self.drug_encoder = Encoder(**kwargs["drug"])
-        self.mlp = self._get_mlp(**kwargs["mlp"])
+        super().__init__()
+        self._determine_feat_method(kwargs["model"]["feat_method"], kwargs["model"]["prot"]["hidden_dim"], kwargs["model"]["drug"]["hidden_dim"])
+        self.prot_encoder = Encoder(**kwargs["model"]["prot"])
+        if kwargs["model"]["drug"]["node"]["module"] == "SweetNet":
```

Comment:
I think I would rather do this a different way - add a `module` entry in the encoder field, which allows you to choose the encoder similarly to how you can choose node embedder/pooler